smart_open/gcs.py

```diff
@@ -0,0 +1,524 @@
+# -*- coding: utf-8 -*-
+"""Implements file-like objects for reading and writing to/from GCS."""
+
+import io
+import logging
+import sys
+
+from google.cloud import storage, exceptions
```

Comment:
Please do this instead:

```suggestion
import google.cloud.exceptions
import google.cloud.storage
```

It makes your code more difficult to write, because it is more verbose.
However, it is easier to read, because the reader doesn't have to remember where `storage` and `exceptions` are coming from.
Readability is more important than writeability in this case.